### ISSUE TITLE
Coreclr workflow instructions incorrect

### ISSUE THREAD

**Issue by benaadams** [dotnet/runtime#33501]
https://github.com/dotnet/runtime/blob/master/docs/workflow/building/coreclr/README.md

It no longer understands `--subsetCategory`
```
❯ .\build  --subsetCategory coreclr -configuration checked
MSBUILD : error MSB1001: Unknown switch.
Switch: --subsetCategory

For switch syntax, type "MSBuild -help"
Build failed.
Some builds failed:
        Configuration: checked, Architecture: x64
```

**Comment by benaadams** [comment]
Single dash now?
```
.\build -subsetCategory coreclr -configuration checked
```

**Comment by AaronRobinsonMSFT** [comment]
Fixed with #34663.
